### PR TITLE
Fix strategy configuration

### PR DESCRIPTION
**Pull request by smcabrera**
This omniauth strategy, like the SurveyMonkey one before it, didn't quite work out of the box. This PR contains the tweaks we needed in order to get it to work in our testing locally

### CODE REVIEW

File: lib/omniauth/strategies/qualtrics.rb [hunk 1]
```diff
@@ -19,26 +18,31 @@ class Qualtrics < OmniAuth::Strategies::OAuth2
...
       info do
         {
-          "url" => access_token.client.site
+          'url' => access_token.client.site
         }
       end
 
+      def callback_url
+        full_host + script_name + callback_path
+      end
```
Comment (smcabrera) [inline lib/omniauth/strategies/qualtrics.rb:27]: We got an error saying it was missing required parameters which is the same error we got with SurveyMonkey so we tried the same remedy and it worked

File: lib/omniauth/strategies/qualtrics.rb [hunk 2]
```diff
@@ -19,26 +18,31 @@ class Qualtrics < OmniAuth::Strategies::OAuth2
...
+
       # Override authorize_params so that we can be deliberate about the value for state
       # and not use the session which is unavailable inside of an iframe for some
       # browsers (ie Safari)
       def authorize_params
         # Only set state if it hasn't already been set
         options.authorize_params[:state] ||= SecureRandom.hex(24)
-        params = options.authorize_params.merge(options_for("authorize"))
+        options.authorize_params[:scope] = 'manage:all'
```
Comment (smcabrera) [inline lib/omniauth/strategies/qualtrics.rb:35]: This is not ideal--we'd rather it have just the scopes that we need, which you _should_ be able to do by setting them in the omniauth config...but I tried that and it didn't work. We can try to look into a better solution in the future.

File: lib/omniauth/strategies/qualtrics.rb (hunk 1 above)
Comment (AlexAlt) [reply lib/omniauth/strategies/qualtrics.rb:27]: is this overriding a method from super or something?

Comment (smcabrera) [reply lib/omniauth/strategies/qualtrics.rb:27]: Correct. I can see if I can find the original explanation of it for SurveyMonkey...

Comment (smcabrera) [reply lib/omniauth/strategies/qualtrics.rb:27]: Here's the same change we made for SurveyMonkey https://github.com/user-interviews/omniauth-surveymonkey-oauth2/pull/1

Based on an identical change for the twitch gem 

image

Here's the original omniauth change that broke things for a lot of gems https://github.com/omniauth/omniauth-oauth2/pull/70

Comment (AlexAlt) [reply lib/omniauth/strategies/qualtrics.rb:27]: twitch integration when

**Review by AlexAlt — APPROVED** [review]
nice- seems like it's battle tested so lgtm

**Comment by smcabrera** [comment]
@michaeldewolf85 we looked at this together so I'm going to assume you're ok with the changes. Also since it's a dependency anyway it's inherently lower risk so I feel confident about merging it in at this point and we can make more changes if we need to

**Pull request merged**